src/newTest/java/org/opensearch/test/framework/TestIndex.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2021-2022 floragunn GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.test.framework;
+
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.action.admin.indices.exists.indices.IndicesExistsRequest;
+import org.opensearch.client.Client;
+import org.opensearch.common.settings.Settings;
+
+public class TestIndex {
```

Comment:
Isn't this just an 'Index', why the prefix of Test?

Reply:
We are following the rule to prefix all test framework related classes to make it very clear, what is test framework and test subject. Of course, we can discuss this rule.

Reply:
What about something along the lines of `IT` (Integration Test) or `TM` (Test Model)?  the `test` prefix is a convention for junit tests cases and makes it harder to distinguish between the test execution systems components and functionality of the product that being exercised

Reply:
Good point. I usually try to avoid acronyms as these can mean many things. However, if you  would prefer either `IT` or `TM` we can of course change it.

Reply:
My preference is towards no distinguishing prefix, but I am comfortable with any approach that you want to take.  Anyone else have opinions?

Reply:
Personally, I would opt for a prefix. There are probably already 10 classes called "Index". For me, navigating to the class I want to see is easier if it has a consistent prefix. I am indifferent regarding what to use exactly.